rasa/model.py

```diff
@@ -314,6 +314,8 @@ async def model_fingerprint(file_importer: "TrainingDataImporter") -> Fingerprin
     stories = await file_importer.get_stories()
     nlu_data = await file_importer.get_nlu_data()
 
+    update_component_indices_of_pipeline(config)
```

Comment:
This is one way of fixing the issue: Add the component indices at this point.

Reply:
I now see that this breaks another test (`test_train.test_model_finetuning_nlu_new_label_already_in_domain`) 🤦‍♂️